### PR TITLE
Add quiet_div, quiet_mod to C backend (Issue #3300)

### PR DESCRIPTION
**Pull request by steven-johnson**
Note: deliberately omitting any tests for this as tests for this intrinsics are missing in general (see Issue #3301)

### CODE REVIEW

File: src/CodeGen_C.cpp [hunk 1]
```diff
@@ -2151,6 +2163,22 @@ void CodeGen_C::visit(const Call *op) {
         user_error << "Signed integer overflow occurred during constant-folding. Signed"
             " integer overflow for int32 and int64 is undefined behavior in"
             " Halide.\n";
+    } else if (op->is_intrinsic(Call::quiet_div)) {
+        internal_assert(op->args.size() == 2);
+        if (is_zero(op->args[1])) {
+            user_error << "Division by constant zero in expression: " << Expr(op) << "\n";
```
Comment (dsharletg) [inline src/CodeGen_C.cpp]: This behavior is different than CodeGen_LLVM... intended?

https://github.com/halide/Halide/blob/760832046efee7587822052fcf0e1b7c3abb02e9/src/CodeGen_LLVM.cpp#L2756

Comment (zvookin) [reply src/CodeGen_C.cpp]: It behaves that way because it is assuming the codegen around it has assured it can never happen.

Comment (steven-johnson) [reply src/CodeGen_C.cpp]: Ah, so in that case it should probably be `internal_error`

Comment (dsharletg) [reply src/CodeGen_C.cpp]: And in CodeGen_LLVM too then?

Comment (steven-johnson) [reply src/CodeGen_C.cpp]: Done

Comment (zvookin) [reply src/CodeGen_C.cpp]: The technique being used in LLVM codegen is what I would expect for some case where the code at runtime can never hit the case where the division by zero executes. Translating that into an internal error requires that the compiler is good enough to remove the IR containing division.

Comment (steven-johnson) [reply src/CodeGen_C.cpp]: ...in which case I should back out the last two changes (and have the C backend just use '0' or other arbitrary for the undef case, I suppose).

**Pull request merged**